x-pack/plugins/spaces/server/lib/create_default_space.ts

```diff
@@ -5,16 +5,23 @@
  */
 
 import { i18n } from '@kbn/i18n';
-// @ts-ignore
-import { getClient } from '../../../../server/lib/get_client_shield';
+
+import { SavedObjectsService } from 'src/legacy/server/kbn_server';
+import { ElasticsearchServiceSetup } from 'src/core/server';
+import { first } from 'rxjs/operators';
 import { DEFAULT_SPACE_ID } from '../../common/constants';
 
-export async function createDefaultSpace(server: any) {
-  const { callWithInternalUser: callCluster } = getClient(server);
+interface Deps {
+  elasticsearch: ElasticsearchServiceSetup;
+  savedObjects: SavedObjectsService;
+}
+
+export async function createDefaultSpace({ elasticsearch, savedObjects }: Deps) {
+  const { getSavedObjectsRepository, SavedObjectsClient } = savedObjects;
 
-  const { getSavedObjectsRepository, SavedObjectsClient } = server.savedObjects;
+  const client = await elasticsearch.dataClient$.pipe(first()).toPromise();
```

Comment:
Is this the right way to get an instance of the client?

Reply:
`dataClient` cannot be update/recreated. @azasypkin correct me if I'm wrong 
https://github.com/elastic/kibana/blob/a0bd3787566b0ade33b64f30927611d459a4f214/src/core/server/elasticsearch/elasticsearch_service.ts#L64

Reply:
> dataClient cannot be update/recreated. @azasypkin correct me if I'm wrong

Yes, that's correct.